src/PowerShellEditorServices/Console/UnixConsoleOperations.cs

```diff
@@ -69,17 +191,67 @@ internal UnixConsoleOperations()
             return false;
         }
 
-        private async Task<bool> SpinUntilKeyAvailable(int millisecondsTimeout, CancellationToken cancellationToken)
+        private async Task<bool> ShortWaitForKeyAsync(CancellationToken cancellationToken)
+        {
+            if (await SpinUntilKeyAvailableAsync(SHORT_READ_TIMEOUT, cancellationToken))
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                return true;
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+            WaitForKeyAvailableAsync = LongWaitForKeyAsync;
+            return false;
+        }
+
+        private bool SpinUntilKeyAvailable(int millisecondsTimeout, CancellationToken cancellationToken)
+        {
+            return SpinWait.SpinUntil(
+                () =>
+                {
+                    s_waitHandle.Wait(30, cancellationToken);
```

Comment:
So here I changed the name of the constants to be more descriptive.  I also switch to Pascal case as these aren't interop constants.  At the time I wrote that I had assumed that because all the constants  I saw happened to be interop that that was the scheme for all constants.

I also added a constant for the 30ms short wait.